### ISSUE TITLE
Feature request: Provide CLI option to enable checking that any non-axiom/extern function or method has a body without requiring compilation

### ISSUE THREAD

**Issue by saltiniroberto** [dafny-lang/dafny#2377]
As far as I know, the only way to force Dafny to check that any function or method without either the `:axiom` or `:extern` attributes has a body is to require compilation.

One of the issues with this is that Dafny will emit an error if one has `:extern` functions or methods for which no external implementation is provided to the CLI.
This is a use case that I have encountered quite often.

It would be great if Dafny allowed checking for bodyless functions or methods without having to go through the compilation phase.

**Comment by keyboardDrummer** [comment]
> One of the issues with this is that Dafny will emit an error if one has :extern functions or methods for which no external implementation is provided to the CLI.

Which error are you referring to? Also, what you mean by providing an external implementation to the CLI?

If I have the program:

```
function method {:extern} ExternallyImplemented(x: int): int

function method ShouldHaveABody(x: int): int

method Main() {
  var x := ExternallyImplemented(3);
  var y := ShouldHaveABody(2);
  print x, y;
}
```

and I run Dafny on it, then it responds:

`Error: Function _module._default.ShouldHaveABody has no body`

So I'm not getting an error for `ExternallyImplemented`, even though I didn't provide an implementation for it on the CLI.

**Comment by saltiniroberto** [comment]
> So I'm not getting an error for ExternallyImplemented, even though I didn't provide an implementation for it on the CLI.

Yep, but if you provide an implementation for it like in the following code snippet
```Dafny
function method {:extern} ExternallyImplemented(x: int): int

function method ShouldHaveABody(x: int): int
{
    x
}

method Main() {
  var x := ExternallyImplemented(3);
  var y := ShouldHaveABody(2);
  print x, y;
}
```
then Dafy does output an error for `ExternallyImplemented`:
```
Errors compiling program into check_for_empty_bodies
(4841,24): error CS0117: '__default' does not contain a definition for 'ExternallyImplemented'
```

What I think would be useful is to have a CLI option that would output error for the code in the comment above (which does not provide the body for `ExternallyImplemented`), but that would not output any error for the code in this post (which provides the body for `ExternallyImplemented`).

**Comment by atomb** [comment]
See #2236, as well. Although that issue focuses on reporting, I've been wondering whether it should include functionality for gating builds based on what would be in the report. (And it'll be able to run without compilation.)
